### DISCUSSION TITLE
Ordering records within expand (relation and back-relations)

### DISCUSSION THREAD

**Discussion by mlusetti** [pocketbase/pocketbase#4769] [Q&A] [answered]
Is there a way to order the fields in the expand section of the records.

I mean, I've a back relation as a posts -> comments where a post field in the comments table  points to the posts table.
I can easly get all comments of a post with the comments_via_post relation but I want that ordered by "comment score" (which is a field in the comments table.

Is that possible?

Previously I've solved it with a view table, now I would like to have it in the JS SDK:

`
const posts = await pb.collection('posts').getFullList({
		filter: 'category="${event.params.id}"',
		expand: 'comments_via_post,
		sort: '-post.post_date,comments_via_post.score',
	});
`

**Comment by ganigeorgiev** [comment, marked as answer]
No, it is not possible. Both `sort` and `filter` applies only to the main targeted collection (in your case `posts`).

**Reply by mlusetti** [reply]
Thanks! There's no problem I can solve my issue with a view.

**Comment by mario125** [comment]
Hello, I am new to PocketBase and I found it challenging to establish the multiple 
relationship between category --> product <-- unit. I managed to display the products
 belonging to each category, but I would also like to add that for each product, 
 the related unit is displayed. It would greatly help me if you could provide some 
 guidelines. These are my tables and the JSON I currently have:

CODE:
```node
const PocketBase = require('pocketbase/cjs')



async function main() {
  try {
    const pb = new PocketBase('XXXXXXXXXXXXXXXXX');

    const userData = await pb
    .collection('users')
    .authWithPassword('XXXXXXXXXXX', 'XXXXXXXXXXXXXXXXX');  

    const productList = await pb.collection("category").getFullList({
   
      expand: "product(category_id)",
    });

  
    console.dir(productList, { depth: null }); 
  } catch (error) {
    console.error('Error:', error);
  }
}

main();


/*
     TABLES:

     category{
      id
      name
     }

     unit{
      id
      name
     }

     product{
      id
      name
      category_id
      unit_id
     }


     RESPONSE:
     [
        {
          collectionId: 'mw04wh002uk1rpn',
          collectionName: 'category',
          created: '2024-04-17 22:18:20.472Z',
          expand: {
            'products(category_id)': [
              {
                category_id: '8022b94juykk4vr',
                collectionId: '1rhk1zpxtfp3vnv',
                collectionName: 'product',
                created: '2024-04-17 22:53:14.152Z',
                id: '7s0c1m6dpvgjfe2',
                name: 'nn',
                unit_id: 'w8s3an6jlmmvrsp',
                updated: '2024-04-18 03:41:01.744Z'
              }
            ]
          },
          id: '8022b94juykk4vr',
          name: 'VERDURAS',
          updated: '2024-04-17 22:18:20.472Z'
        },
        {
          collectionId: 'mw04wh002uk1rpn',
          collectionName: 'category',
          created: '2024-04-17 22:47:52.047Z',
          expand: {
            'products(category_id)': [
              {
                category_id: 'i5848eqkt3o5vb4',
                collectionId: '1rhk1zpxtfp3vnv',
                collectionName: 'product',
                created: '2024-04-17 22:53:54.828Z',
                id: 't9xugad29ow426v',
                name: 'fgfg',
                unit_id: 'w8s3an6jlmmvrsp',
                updated: '2024-04-18 03:40:55.093Z'
              },
              {
                category_id: 'i5848eqkt3o5vb4',
                collectionId: '1rhk1zpxtfp3vnv',
                collectionName: 'product',
                created: '2024-04-18 00:56:08.223Z',
                id: '5iu05k9j012jkcw',
                name: 'pollo',
                unit_id: 'w8s3an6jlmmvrsp',
                updated: '2024-04-18 01:13:29.870Z'
              }
            ]
          },
          id: 'i5848eqkt3o5vb4',
          name: 'CARNES',
          updated: '2024-04-17 22:47:52.047Z'
        }
      ]
     
     
     */

**Reply by ganigeorgiev** [reply]
@mario125 I don't understand what actually is the question, nor how it is related to the above discussion. For working with relations, please refer to https://pocketbase.io/docs/working-with-relations/.

**Comment by hexxyan** [comment]
> No, it is not possible. Both `sort` and `filter` applies only to the main targeted collection (in your case `posts`).

Sorry for the intrusion，is it possible to customize the default sorting for each collection？Rather than specifying sort at params, can it be predefined? This way, the JSON obtained through 'expand' (relation and back-relations) would also follow the predefined sorting.
I suppose this feature would be well-suited to be added in the “API rules” section, where it could determine the default sorting.
